src/reconcile/rec_row.c

```diff
@@ -628,6 +628,30 @@ __rec_row_leaf_insert(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins)
     return (0);
 }
 
+/*
+ * __rec_simplify_time_window --
+ *     Where possible modify time window values to avoid writing them to the cell later.
+ */
+static inline void
+__rec_simplify_time_window(WT_SESSION_IMPL *session, WT_TIME_WINDOW *tw)
+{
+    wt_timestamp_t pinned_ts;
+    /*
+     * Get a copy of the pinned timestamp and oldest ID. It is OK if these are out of date, they
+     * only move forward, so the worst case is that we store some information that is no longer
+     * relevant.
+     */
+    __wt_txn_pinned_timestamp(session, &pinned_ts);
+    if (!tw->prepare) {
+        if (tw->stop_txn == WT_TXN_MAX && tw->start_txn < __wt_txn_oldest_id(session))
+            tw->start_txn = WT_TXN_NONE;
+
+        if (tw->durable_start_ts < pinned_ts && tw->stop_ts == WT_TS_MAX &&
+          tw->start_ts < pinned_ts)
+            tw->start_ts = WT_TS_NONE;
```

Comment:
I think there is a possibility that we store a durable timestamp when we don't expect to because of this check, which I suspect could account for the rollback to stable test failure.

Reply:
The rollback to stable failure appears to be because we're moving our start_txn before the checkpoint_id. Which I think make us visible mid checkpoint causing weirdness in rollback to stable.

Reply:
Additionally the durable stop and start timestamps are both initialized to WT_TS_NONE so if I set them to that I think it messes with the logic in rollback to stable which treats them as unset. However I haven't yet fully investigated rollback to stables handling of the durable start and stop ts.

Reply:
Well I think my original guess about it being related to checkpoint wasn't correct so I'll continue looking into it.

Reply:
I'd structure this check like:
```
/* Avoid retrieving the pinned timestamp unless we need it. */
if (tw->stop_ts == WT_TS_MAX) {
    /* The durable stop timestamp should be it's default value whenever the stop timestamp is */
    WT_ASSERT(session, tw->durable_stop_ts == WT_TS_NONE);
    __wt_txn_pinned_timestamp(session, &pinned_ts);
    /* The durable start timestamp is always greater than or equal to the start timestamp, so we only need to check the more conservative one here. */
    WT_ASSERT(session, tw->start_ts <= tw->durable_start_ts);
    if (tw->start_ts < pinned_ts)
        tw->start_ts = tw->durable_start_ts = WT_TS_NONE;
}
```

Reply:
Generally I think this should move to timestamp.i as well, and be called "__wt_time_window_clear_obsolete" or similar.

Reply:
Added suggested code and moved it to timestamp.i

Reply:
So the in memory version was rolling back more updates in rollback to stable than it should've been, what was happening was that when we reconciled the page the disk image was still created and the txn id was set to WT_TXN_NONE. However the in memory update chain was still persisted as we're running in memory. So when we'd come back through reconciliation we'd append a new update to the end of the update chain in `__rec_append_orig_value` as it didn't match any of the in memory updates as its txn_id was reset.